core/src/main/scala/sttp/client3/logging/Log.scala

```diff
@@ -39,30 +39,59 @@ class DefaultLog[F[_]](
 ) extends Log[F] {
 
   def beforeRequestSend(request: Request[_, _]): F[Unit] =
+    request.loggingOptions match {
+      case Some(options) => before(request, options.logRequestBody, options.logRequestHeaders)
```

Comment:
hm question is, should `options.logRequestBody` (and others) always override what's in the `DefaultLog` options? Maybe the fields in `LoggingOptions` should be optional, where a `None` means "keep default"

Reply:
hmm, why not I options are optional themselves IMO there is no need for adding one more layer of optionality here.
Additionally loggingOptions have to be explicitly added by user so no surprises here

Reply:
the idea would be to be able to override only a single setting (e.g. `logRequestHeaders`), while leaving others intact (as defined in the wrapper backend)

Reply:
you are right , will implement